### PR TITLE
W-15346051: fix startup failure on windows

### PR DESCRIPTION
**Pull request by akshaysonvane**

### CODE REVIEW

**Comment by akshaysonvane** [comment]
--validate

File: mule-wrapper-additional-parameters-parser/src/main/java/org/mule/runtime/params/AdditionalJvmParameters.java [hunk 1]
```diff
@@ -264,10 +267,18 @@ protected static void writeWrapperProps(String[] args, FileWriter writer) throws
...
     for (Entry entry : bootstrapProperties.entrySet()) {
-      if (entry.getKey().toString().matches("wrapper\\.java\\.additional\\.<n\\d>")) {
-        writer.write(wrapperPrefix + ++paramIndex + "=" + entry.getValue().toString() + "\n");
+      Matcher additionalMatcher = additionalPattern.matcher(entry.getKey().toString());
+      if (additionalMatcher.matches()) {
+        String wrapperSuffix = additionalMatcher.group(2) != null ? additionalMatcher.group(2) : "";
+        int index = nComponentIndexMap.computeIfAbsent(additionalMatcher.group(1), k -> ++paramIndex);
+
+        writer.write(wrapperPrefix + index + wrapperSuffix + "=" + entry.getValue().toString() + "\n");
       } else if (entry.getKey().toString().matches("wrapper\\.java\\.classpath\\.<n\\d>")) {
```
Comment (akshaysonvane) [inline mule-wrapper-additional-parameters-parser/src/main/java/org/mule/runtime/params/AdditionalJvmParameters.java:282]: Not adding a similar `indexMap` for `wrapper.java.classpath` as these properties [don't](https://wrapper.tanukisoftware.com/doc/english/prop-java-classpath-n.html) have a suffix.

**Review by asanguinetti — CHANGES_REQUESTED** [review]

File: mule-wrapper-additional-parameters-parser/src/test/java/org/mule/runtime/params/AdditionalJvmParametersTestCase.java [hunk 2]
```diff
@@ -216,4 +220,33 @@ public void testAdditionalJvmParameters() throws IOException {
...
+    String prop1 = list.get(0);
+    int prop1EndIdx = prop1.indexOf("=", wrapperPrefix.length() + 1);
+    String prop2 = list.get(1);
+    int prop2EndIdx = prop2.indexOf(".", wrapperPrefix.length() + 1);
+    assertThat(prop1.substring(0, prop1EndIdx), is(prop2.substring(0, prop2EndIdx)));
+
+    Arrays.asList("wrapper.java.additional.\\d+=value2",
+                  "wrapper.java.classpath.\\d+=classpathValue1",
+                  "other.property=otherValue")
+        .forEach(pattern -> assertThat(actualList.stream().anyMatch(p -> p.matches(pattern)), is(true)));
```
[the remark this answers is not in the document]
Comment (akshaysonvane) [reply mule-wrapper-additional-parameters-parser/src/test/java/org/mule/runtime/params/AdditionalJvmParametersTestCase.java]: Good point. Updated!

File: mule-wrapper-additional-parameters-parser/src/test/java/org/mule/runtime/params/AdditionalJvmParametersTestCase.java [hunk 3]
```diff
@@ -216,4 +221,31 @@ public void testAdditionalJvmParameters() throws IOException {
...
+    properties.setProperty("wrapper.java.additional.<n1>.suffix", "value1");
+    properties.setProperty("wrapper.java.additional.<n2>", "value2");
+    properties.setProperty("wrapper.java.classpath.<n1>", "classpathValue1");
+    properties.setProperty("other.property", "otherValue");
+
+    AdditionalJvmParameters.processBootstrapProperties(properties, writer);
+
+    List<String> actualList = asList(writer.toString().split("\n"));
+    List<String> list = actualList.stream().filter(s -> s.matches("wrapper\\.java\\.additional.*value1")).collect(toList());
+    assertThat(list.size(), is(2));
```
Comment (asanguinetti) [inline mule-wrapper-additional-parameters-parser/src/test/java/org/mule/runtime/params/AdditionalJvmParametersTestCase.java:239]: Sorry I missed this one in the previous review. We can also use a better matcher here.

**Review by asanguinetti — CHANGES_REQUESTED** [review]

**Review by asanguinetti — APPROVED** [review]

**Pull request merged**